### PR TITLE
Update OSX build scripts and associated data files.

### PR DESCRIPTION
**Pull request by jdykstra**
Update the build scripts and associated data used to build QL for the Mac platform.  The updated scripts, when used with XCode 15.0.1 or later, build an app that will run on MacOS Sonoma and earlier (and with luck, later).

This fixes issues #4384, #4385 and #4389.

New code copyright 2023 by John Dykstra.
This program is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation; either version 2 of the License, or (at your option) any later version.

Check-list
----------

 * [x] There is a linked issue discussing the motivations for this feature or bugfix
 * [ na ] Unit tests have been added where possible
 * [ na ] I've added / updated documentation for any user-facing features.
 * [ na ] Performance seems to be comparable or better than current `main`

What this change is adding / fixing
-----------------------------------
Catch up on technical debt in dev-utils/osx_bundle so that QL can be built for Macintosh.  If this is done with Xcode 15.0.1 or later, the resulting build products run properly on the latest MacOS release, Sonoma, as well as earlier releases.  At least four beta testers are running this code on Sonoma without any reported issues.

A build based on version-4.6.0 (numbered 4.6.1) is at https://github.com/jdykstra/quodlibet/releases/tag/release-4.6.1.
This may be added to the QL upload page for Mac users to use until the next formal QL release.

NOTE:  The ExFalso currently built by these scripts will not work properly.  See issue #1 in jdykstra/quodlibet, which I will reopen against quodlibet/quodlibet once this commit is integrated.  A proper fix to this issue will add significant complexity to the launch scripts, and may be unnecessary, since Apple has signaled that the root cause of this issue is a bug that they will fix.

### CODE REVIEW

File: quodlibet/_main.py [hunk 1]
```diff
@@ -206,9 +206,9 @@ def get_build_description():
 
         if build.BUILD_INFO:
             notes.append(build.BUILD_INFO)
-
-    version_string = ".".join(map(str, version))
-    note = " (%s)" % ", ".join(notes) if notes else ""
+    
+    version_string = u".".join(map(str, version))
```
Comment (declension) [inline quodlibet/_main.py:210]:
```suggestion
    version_string = ".".join(map(str, version))
```
Unicode literal syntax is long deprecates and Ruff now errors on this

File: quodlibet/_main.py [hunk 2]
```diff
@@ -206,9 +206,9 @@ def get_build_description():
...
+    note = u" (%s)" % u", ".join(notes) if notes else u""
```
Comment (declension) [inline quodlibet/_main.py:211]:
```suggestion
    note = f" ({','.join(notes)})" if notes else ""
```

**Comment by declension** [comment]
> Ruff tests are failing on several platforms, but I believe this is unrelated to the current commit.

Yes, sorry about this. [All green now in `main`](https://github.com/quodlibet/quodlibet/actions/runs/7019679031/job/19097661091). Can you try merging / rebasing from there?

File: quodlibet/_main.py (hunk 2 above)
Comment (jdykstra) [reply quodlibet/_main.py]: Fixed

**Comment by jdykstra** [comment]
Comments resolved and all tests completed successfully.  Thanks!

**Comment by declension** [comment]
Thanks @jdykstra !

**Comment by oaubert** [comment]
Thanks a lot for this work, I have been able to base my work on it and adapt it to Advene (same Python/Gtk/Gstreamer stack basically) so that it runs again on Sonoma. I also managed to get it built on Github runners, which required mainly to get new versions of some packages, as defined in modulesets/quodlibet.modules. If  you somehow need to do this, you can get the Advene version of the configuration files/scripts at https://github.com/oaubert/advene/tree/master/dev/osx_bundle

**Comment by jdykstra** [comment]
> I also managed to get it built on Github runners, which required mainly to get new versions of some packages, as defined in modulesets/quodlibet.modules. If you somehow need to do this, you can get the Advene version of the configuration files/scripts at https://github.com/oaubert/advene/tree/master/dev/osx_bundle.

Great!  Thank you, Oliver!

**Pull request merged**